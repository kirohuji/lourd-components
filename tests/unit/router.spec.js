import { shallowMount, createLocalVue } from "@vue/test-utils";
import VueRouter from "vue-router";
const App = {
  render() {
    return (
      <div id="app">
        <h1>Hello App!</h1>
        <p>
          <router-link to="/foo">Go to Foo</router-link>
          <router-link to="/bar">Go to Bar</router-link>
        </p>
        <router-view></router-view>
      </div>
    );
  },
};
const Foo = { template: "<div>foo</div>" };
const Bar = { template: "<div>bar</div>" };
const routes = [
  { path: "/foo", component: Foo },
  { path: "/bar", component: Bar },
];

describe("App", () => {
  it("with-router", () => {
    const localVue = createLocalVue();
    localVue.use(VueRouter);
    const router = new VueRouter({
      routes,
    });
    const wrapper = shallowMount(App, {
      localVue,
      router,
    });
    expect(wrapper).toMatchSnapshot();
  });
});
