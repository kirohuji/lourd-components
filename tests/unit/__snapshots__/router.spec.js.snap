// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App with-router 1`] = `
<div id="app">
  <h1>Hello App!</h1>
  <p>
    <router-link-stub to="/foo" tag="a" ariacurrentvalue="page" event="click">Go to Foo</router-link-stub>
    <router-link-stub to="/bar" tag="a" ariacurrentvalue="page" event="click">Go to Bar</router-link-stub>
  </p>
  <router-view-stub name="default"></router-view-stub>
</div>
`;
